setuptools/tests/test_build_meta.py

```diff
@@ -442,6 +443,19 @@ def test_sys_argv_passthrough(self, tmpdir_cwd):
         with pytest.raises(AssertionError):
             build_backend.build_sdist("temp")
 
+    @pytest.mark.parametrize('build_hook', ('build_sdist', 'build_wheel'))
+    def test_build_with_empty_setuppy(self, build_backend, build_hook):
+        files = {'setup.py': ''}
+        path.build(files)
+
+        with pytest.raises(
+                ValueError,
+                match=re.escape(
+                    'No distribution was found. The distribution was '
+                    'possibly not built. Ensure that your `setup.py` '
+                    'is not empty and that it calls `setup()`.')):
```

Comment:
Simplify match to avoid over-matching the expected text. Just assert on the gist of the message.

```suggestion
                match=re.escape('No distribution was found.')):
```